phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/SearchBox.java

```diff
@@ -1,24 +1,29 @@
 package org.phoenicis.javafx.views.mainwindow.ui;
 
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
 import org.phoenicis.javafx.views.common.ThemeManager;
 
 import javafx.beans.property.StringProperty;
 import javafx.scene.control.TextField;
 import javafx.scene.control.Button;
 import javafx.scene.layout.AnchorPane;
 
+import java.util.function.Consumer;
+
 public class SearchBox extends AnchorPane {
     private TextField searchField;
     private Button clearButton;
 
     private static final String cleanButtonIcon = "icons/mainwindow/general/edit-clear.png";
 
-    public SearchBox(ThemeManager themeManager) {
+    public SearchBox(ThemeManager themeManager, Consumer<String> onSearch, Runnable onClear) {
         this.searchField = new TextField();
 
         this.searchField.getStyleClass().add("searchBar");
         this.searchField.prefHeightProperty().bind(this.prefHeightProperty());
         this.searchField.prefWidthProperty().bind(this.prefWidthProperty());
+        this.searchField.setOnAction(event -> onSearch.accept(getText()));
```

Comment:
I think we should keep the delayed search. It might not be noticeable when the search is performed with every new/removed character on a reasonably fast computer. However, you might notice it on a slow computer. Moreover it is useless computation.